### DISCUSSION TITLE
Add support for Java 17

### DISCUSSION THREAD

**Discussion by dsyer** [community/community#6984] [Codespaces] [answered]
Java 17 is the new LTS, but it doesn't show as a choice when you create a devcontainer. Ideally Codespaces shouldn't even need me to create a devcontainer - it would be obvious from the source code that Java 17 is required, but probably it should be the default anyway IMO.

**Comment by Chuxel** [comment, marked as answer]
While you can technically opt to use another image, for pre-built images, the VS Code extension for Java did initially not have JDK 17 support when JDK was first released, so we did not want to set false expectations. That has since changed and you'll find 17 in the configuration options now in that UX.

The default codespaces image updating its auto-detection makes sense, but is not yet in place.

**Comment by dsyer** [comment]
Thanks. There seem to be some issues with the autodetection still (e.g. https://github.com/microsoft/vscode-dev-containers/issues/1245) but probably not directly relevant.
